oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/file/tar/SegmentTarWriter.java

```diff
@@ -93,7 +94,7 @@ public SegmentTarWriter(File file, FileStoreMonitor monitor, IOMonitor ioMonitor
     }
 
     @Override
-    public void writeSegment(long msb, long lsb, byte[] data, int offset, int size, int generation, int fullGeneration, boolean compacted) throws IOException {
+    public void writeSegment(long msb, long lsb, byte @NotNull [] data, int offset, int size, int generation, int fullGeneration, boolean compacted) throws IOException {
```

Comment:
Please refrain from adding unrelated changes to the PR. The additional annotations are justified, but should be handled with a dedicated JIRA ticket / PR.